静态链接到opencc时报错符号找不到

### 描述问题

静态链接到opencc时报错符号找不到。

### 期待的结果

静态链接到opencc能正常完成。

### 错误信息

如果是xmake相关编译问题，请加上`-vD`参数运行，并给出完整编译输出信息以及执行的完整命令，例如：

```console
$ xmake -vD
error: @programdir\modules\private\async\runjobs.lua:232: @programdir\actions\build\kinds\binary.lu                    ua:73: @programdir\modules\core\tools\link.lua:159: main.cpp.obj : error LNK2019: 无法解析的外部符号 "__declspec(dllimport) public: __cdecl opencc::Config::Config(void)" (__imp_??0Config@opencc@@QEAA@XZ)，函数 main 中引用了该符号
main.cpp.obj : error LNK2019: 无法解析的外部符号 "__declspec(dllimport) public: virtual __cdecl opencc::Config::~Config(build\windows\x64\release\test.exe : fatal error LNK1120: 2 个无法解析的外部命令
```

### 相关环境

请提供编译和运行环境信息，下面是一些必须填写的基础信息，便于我们针对性排查问题：

- xmake版本：2.5.5
- xmake运行平台：Windows
- xmake目标平台：Windows

### 其他信息

请提供关键的 xmake.lua 配置内容，或者完整 xmake.lua，也可以是其他附加信息帮助我们诊断问题(比如截图，xmake.lua或者可复现的demo)，以及你遇到的问题的一些背景信息。

<!--
如果想进一步探讨xmake相关问题，可以加入我们的技术社区互相交流：

* 社区：[Reddit论坛](https://www.reddit.com/r/xmake/)
* 聊天：[Telegram群组](https://t.me/tbooxorg), [Gitter聊天室](https://gitter.im/xmake-io/xmake)
* QQ群：343118190(技术支持), 662147501
* 微信公众号：tboox-os （可通过此公众号申请加入我们的微信群）
-->

--- a/packages/o/opencc/xmake.lua
+++ b/packages/o/opencc/xmake.lua
@@ -17,6 +17,9 @@ package("opencc")
         if package:is_plat("linux", "mingw") and not package:config("shared") then
             package:add("links", "opencc", "marisa")
         end
+        if not package:config("shared") then
+            package:add("defines", "Opencc_BUILT_AS_STATIC")
+        end
         package:addenv("PATH", "bin")
     end)
 
@@ -29,4 +32,9 @@ package("opencc")
 
     on_test("windows", "mingw@windows,msys", "linux", "macosx", "bsd", function (package)
         assert(package:has_cfuncs("opencc_open", {includes = "opencc/opencc.h"}))
+        assert(package:check_cxxsnippets({test = [[
+            void test() {
+                opencc::Config config;
+            }
+        ]]}, {includes = {"opencc/Config.hpp"}}))
     end)

